sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -789,6 +793,36 @@ case class Range(
     }
   }
 
+  private def getHistogramStatistics = {
+    val numBins = conf.histogramNumBins
+    val height = numElements.toDouble / numBins
+    val percentileArray = (0 to numBins).map(i => i * height).toArray
+
+    var binId = 0
+    val binArray = new Array[HistogramBin](numBins)
+    var lowerIndex = percentileArray(0)
+    var lowerBinValue = getRangeValue(0)
+    while (binId < numBins) {
+      val upperIndex = percentileArray(binId + 1)
+      val upperBinValue = getRangeValue(math.max(math.ceil(upperIndex).toInt - 1, 0))
+      val ndv = math.max(math.ceil(upperIndex).toInt - math.ceil(lowerIndex).toInt, 1)
```

Comment:
Please do not repeat `math.ceil(upperIndex)` here.

Reply:
Yes, updated